Feature/hfhub utils

ivelin opened the pull request:
HuggingFace Hub is a popular cloud service for storing open source and private models, datasets and demo spaces. It would make it easier for timeseries ML engineers to build and deploy demos of Darts models and timeseries datasets if they can be saved and loaded from hfhub.

Notebook with example usage:
https://github.com/ivelin/canswim/blob/19bd79d9e0c80b982da15518833fc583c1f9a5ab/canswim_tests.ipynb

ivelin committed:
Create hfhub.py util for HugginfFace Hub model storage

HFHub is a convenient cloud service for storing models, datasets and demo spaces. It would make it easier for timeseries ML engineers to build and deploy demos of Darts models and timeseries datasets if they can be saved and loaded from hfhub.

Example usage (see cell 56):
https://github.com/ivelin/canswim/blob/pypackage/model_sandbox.ipynb

ivelin committed:
Update CHANGELOG.md

ivelin committed:
Update CHANGELOG.md

ivelin committed:
Update CHANGELOG.md

add timeseries info

ivelin committed:
add timeseries upload and download methods

VascoSch92 commented on darts/utils/hfhub.py:
```diff
@@ -0,0 +1,93 @@
+                local_dir=tmpdirname,
+                token=self.HF_TOKEN,
+            )
+            print(os.listdir(tmpdirname))
```
can not use a log instead of a print statement?

ivelin replied:
Agreed. Updated.

ivelin committed:
replace print with logger

Addressing review comment by @VascoSch92
https://github.com/unit8co/darts/pull/2201#discussion_r1474996216

ivelin committed:
add token param to create_repo

dennisbader commented:
Closing this one, since we are not planning on integrating this at least not in the near future.
